packetbeat/protos/pgsql/parse.go

```diff
@@ -0,0 +1,567 @@
+package pgsql
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+var (
+	errInvalidString     = errors.New("invalid pgsql string")
+	errEmptyFieldsBuffer = errors.New("empty fields buffer")
+	errNoFieldName       = errors.New("can not read column field")
+	errFieldBufferShort  = errors.New("field buffer to small for field count")
+	errFieldBufferBig    = errors.New("field count to small for field buffer size")
+)
+
+func (pgsql *Pgsql) pgsqlMessageParser(s *PgsqlStream) (bool, bool) {
+	debugf("parseMessage, off=%v", s.parseOffset)
+
+	var ok, complete bool
+
+	switch s.parseState {
+	case PgsqlStartState:
+		ok, complete = pgsql.parseMessageStart(s)
+	case PgsqlGetDataState:
+		ok, complete = pgsql.parseMessageData(s)
+	default:
+		logp.Critical("Pgsql invalid parser state")
+	}
+
+	detailedf("parseMessage return: ok=%v, complete=%v, off=%v",
```

Comment:
actually no. it was for wrapping the switch statements, but I see what you mean. Will update the output.